bloom_nofos/nofos/test_nofo.py

```diff
@@ -6815,3 +6964,132 @@ def test_mixed_match_and_nonmatch(self):
+        self.subsection_1.save()
+        self.subsection_2.body = "Apply for hrsa-24-019 now"
+        self.subsection_2.save()
+        updated = replace_value_in_subsections(
+            [self.subsection_1.id, self.subsection_2.id],
+            "HRSA-24-019",
+            "HRSA-24-020",
+        )
+        self.assertEqual(len(updated), 2)
+        self.subsection_1.refresh_from_db()
+        self.subsection_2.refresh_from_db()
+        self.assertIn("HRSA-24-020", self.subsection_1.body)
+        self.assertIn("HRSA-24-020", self.subsection_2.body)
+        self.assertNotIn("HRSA-24-019", self.subsection_1.body)
+        self.assertNotIn("hrsa-24-019", self.subsection_2.body)
+
+    def test_replace_number_with_hyphens_and_spaces(self):
+        self.subsection_1.body = "Apply to HRSA - 24 - 019"
+        self.subsection_1.save()
+        updated = replace_value_in_subsections(
+            [self.subsection_1.id],
+            "HRSA-24-019",
+            "HRSA-24-020",
+        )
+        self.assertEqual(len(updated), 0)
+        self.subsection_1.refresh_from_db()
+        self.assertIn("HRSA - 24 - 019", self.subsection_1.body)
+
+    def test_replace_multiple_occurrences_in_single_subsection(self):
+        self.subsection_1.body = "Test NOFO is great. I love Test NOFO!"
+        self.subsection_1.save()
+        updated = replace_value_in_subsections(
+            [self.subsection_1.id],
+            "Test NOFO",
+            "New Title",
+        )
+        self.assertEqual(len(updated), 1)
+        self.subsection_1.refresh_from_db()
+        self.assertIn("New Title is great. I love New Title!", self.subsection_1.body)
+        self.assertNotIn("Test NOFO", self.subsection_1.body)
+
+    def test_replace_with_empty_value(self):
+        self.subsection_1.body = "The title is Test NOFO"
+        self.subsection_1.save()
+        updated = replace_value_in_subsections(
+            [self.subsection_1.id],
+            "Test NOFO",
+            "",
+        )
+        self.assertEqual(len(updated), 1)
+        self.subsection_1.refresh_from_db()
+        self.assertIn("The title is ", self.subsection_1.body)
+        self.assertNotIn("Test NOFO", self.subsection_1.body)
+
+    # NOTE - Commenting our for the time being. This feels like we should
+    # conditionally check this in the function behavior.
+    # def test_replace_empty_old_value_behavior(self):
+    #     # Since the function appears to process empty strings (based on the test failure),
+    #     # we should test the actual behavior rather than what we initially expected
+    #     original_body = self.subsection_1.body
+    #     updated = replace_value_in_subsections(
+    #         [self.subsection_1.id],
+    #         "",
+    #         "New Value",
+    #     )
+    #     self.assertEqual(len(updated), 1)
+    #     self.subsection_1.refresh_from_db()
+    #     self.assertNotEqual(self.subsection_1.body, original_body)
```

Comment:
Interesting, are you saying this will run the `replace` with an empty string (effectively, erasing the original value)?